JupyterLab 3.4 build is failing on `crypto` in the same way as issue 11854

## Description

`jupyter lab build` eventually fails with:

```
$ webpack --config webpack.prod.config.js
[webpack-cli] ModuleNotFoundError: Module not found: Error: Can't resolve 'crypto' in '/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:

    add a fallback 'resolve.fallback: { "crypto": require.resolve("crypto-browserify") }'
    install 'crypto-browserify'
    If you don't want to include a polyfill, you can use an empty module like this:
    resolve.fallback: { "crypto": false }
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/Compilation.js:1768:28
```

This looks functionally identical to #11854 (and indeed there has been a recent comment, but since that's a closed issue I figured opening a new one would be better).


## Reproduce

Well, in *my* environment, which is a complicated mostly-conda one, `jupyter lab build`.  But obviously the project has tested THAT.  I have a lot of prebuilt modules installed, and what I know right now is that if `jupyter lab build` is run, even without any of my non-conda and non-prebuilt extensions, `jupyter lab build` fails like this.  I have not yet eliminated all of the prebuilt modules.

My build process uses the Dockerfile generated by running GNU Make in a checked-out copy of:

https://github.com/lsst-sqre/sciplat-lab

I'm currently testing with the parameters `tag=w_2022_18 supplementary=jl34` but it's not underlying-version-of-the-stack-container sensitive.

## Expected behavior

`jupyter lab build` should succeed.

## Context


- Operating System and version: derived from CentOS 7.  
- JupyterLab version: 3.4.0

<!--The more content you provide, the more we can help!-->

<details><summary>Troubleshoot Output</summary>
Doesn't fit in the 64K the body allows me.
</details>


<details><summary>Command Line Output</summary>
<pre>
(lsst-scipipe-3.0.0) [root@72d8a7ca7699 stack]# jupyter lab build --debug
[LabBuildApp] Searching ['/opt/lsst/software/stack', '/root/.jupyter', '/root/.local/etc/jupyter', '/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[LabBuildApp] Looking for jupyter_config in /etc/jupyter
[LabBuildApp] Looking for jupyter_config in /usr/local/etc/jupyter
[LabBuildApp] Looking for jupyter_config in /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/etc/jupyter
[LabBuildApp] Looking for jupyter_config in /root/.local/etc/jupyter
[LabBuildApp] Looking for jupyter_config in /root/.jupyter
[LabBuildApp] Looking for jupyter_config in /opt/lsst/software/stack
[LabBuildApp] Looking for JupyterLab_config in /etc/jupyter
[LabBuildApp] Looking for JupyterLab_config in /usr/local/etc/jupyter
[LabBuildApp] Looking for JupyterLab_config in /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/etc/jupyter
[LabBuildApp] Looking for JupyterLab_config in /root/.local/etc/jupyter
[LabBuildApp] Looking for JupyterLab_config in /root/.jupyter
[LabBuildApp] Looking for JupyterLab_config in /opt/lsst/software/stack
[LabBuildApp] JupyterLab 3.4.0
[LabBuildApp] Building in /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab
[LabBuildApp] Node v17.9.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/staging/yarn.js install --non-interactive
/[LabBuildApp] yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.55s.

[LabBuildApp] > node /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/staging/yarn.js yarn-deduplicate -s fewer --fail
-[LabBuildApp] yarn run v1.21.1
$ /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/.bin/yarn-deduplicate -s fewer --fail
Done in 0.36s.

[LabBuildApp] > node /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/staging/yarn.js run build:prod:minimize
-[LabBuildApp] yarn run v1.21.1
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] ModuleNotFoundError: Module not found: Error: Can't resolve 'crypto' in '/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "crypto": require.resolve("crypto-browserify") }'
	- install 'crypto-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "crypto": false }
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/Compilation.js:1768:28
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:742:13
    at eval (eval at create (/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:276:22
    at eval (eval at create (/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:405:22
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:119:11
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:650:24
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:813:8
    at /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/webpack/lib/NormalModuleFactory.js:933:5
resolve 'crypto' in '/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/dist'
  Parsed request is a module
  using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/package.json (relative path: ./dist)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/dist/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/coreutils/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/@lumino/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules
        single file module
          using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/package.json (relative path: ./node_modules/crypto)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto is not a file
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.js doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.json doesn't exist
            .wasm
              Field 'browser' doesn't contain a valid alias configuration
              /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.wasm doesn't exist
        existing directory /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto
          using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/package.json (relative path: .)
            using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/package.json (relative path: ./node_modules/crypto)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto is not a file
              .js
                Field 'browser' doesn't contain a valid alias configuration
                /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto.wasm doesn't exist
              as directory
                existing directory /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto
                  using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/package.json (relative path: .)
                    use ./index.js from main in package.json
                      using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/package.json (relative path: ./index.js)
                        no extension
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js doesn't exist
                        .js
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js.js doesn't exist
                        .json
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js.json doesn't exist
                        .wasm
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js.wasm doesn't exist
                        as directory
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js doesn't exist
                    using path: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index
                      using description file: /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/package.json (relative path: ./index)
                        no extension
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index doesn't exist
                        .js
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.js doesn't exist
                        .json
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.json doesn't exist
                        .wasm
                          Field 'browser' doesn't contain a valid alias configuration
                          /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/staging/node_modules/crypto/index.wasm doesn't exist
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/lab/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/jupyter/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/share/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/conda/node_modules doesn't exist or is not a directory
      /opt/lsst/software/stack/node_modules doesn't exist or is not a directory
      /opt/lsst/software/node_modules doesn't exist or is not a directory
      /opt/lsst/node_modules doesn't exist or is not a directory
      /opt/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
Build failed.
Troubleshooting: If the build failed due to an out-of-memory error, you
may be able to fix it by disabling the `dev_build` and/or `minimize` options.

If you are building via the `jupyter lab build` command, you can disable
these options like so:

jupyter lab build --dev-build=False --minimize=False

You can also disable these options for all JupyterLab builds by adding these
lines to a Jupyter config file named `jupyter_config.py`:

c.LabBuildApp.minimize = False
c.LabBuildApp.dev_build = False

If you don't already have a `jupyter_config.py` file, you can create one by
adding a blank file of that name to any of the Jupyter config directories.
The config directories can be listed by running:

jupyter --paths

Explanation:

- `dev-build`: This option controls whether a `dev` or a more streamlined
`production` build is used. This option will default to `False` (i.e., the
`production` build) for most users. However, if you have any labextensions
installed from local files, this option will instead default to `True`.
Explicitly setting `dev-build` to `False` will ensure that the `production`
build is used in all circumstances.

- `minimize`: This option controls whether your JS bundle is minified
during the Webpack build, which helps to improve JupyterLab's overall
performance. However, the minifier plugin used by Webpack is very memory
intensive, so turning it off may help the build finish successfully in
low-memory environments.

[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/labapp.py", line 196, in start
    raise e

[LabBuildApp]   File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/labapp.py", line 187, in start
    build(

[LabBuildApp]   File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/commands.py", line 498, in build
    return handler.build(

[LabBuildApp]   File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/jupyterlab/commands.py", line 713, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

An error occurred.
RuntimeError: JupyterLab failed to build
See the log file for details:  /tmp/jupyterlab-debug-t083bbe3.log
[LabBuildApp] Exiting application: JupyterLab
</pre>
</details>

--- a/builder/src/webpack.config.base.ts
+++ b/builder/src/webpack.config.base.ts
@@ -97,6 +97,7 @@ module.exports = {
     fallback: {
       url: false,
       buffer: false,
+      crypto: false,
       // See https://github.com/webpack/webpack/blob/3471c776059ac2d26593ea39f9c47c1874253dbb/lib/ModuleNotFoundError.js#L13-L42
       path: require.resolve('path-browserify'),
       process: require.resolve('process/browser')

